engine/src/test/java/com/datatorrent/stram/util/LoggerUtilTest.java

```diff
@@ -16,86 +16,153 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.log4j;
+package com.datatorrent.stram.util;
 
-import java.lang.reflect.Field;
 import java.util.Map;
 
-import org.junit.Assert;
 import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
 import org.junit.Test;
+import org.junit.runners.MethodSorters;
 import org.slf4j.LoggerFactory;
 
+import org.apache.log4j.Category;
+import org.apache.log4j.Level;
+import org.apache.log4j.LogManager;
+
 import com.google.common.collect.Maps;
 
-import com.datatorrent.stram.StreamingAppMaster;
-import com.datatorrent.stram.api.StramEvent;
-import com.datatorrent.stram.client.DTConfiguration;
-import com.datatorrent.stram.engine.StreamingContainer;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
 
-public class DTLoggerFactoryTest
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class LoggerUtilTest
 {
+  private static final org.slf4j.Logger logger = LoggerFactory.getLogger(LoggerUtilTest.class);
 
   @BeforeClass
   public static void setup() throws Exception
   {
-    System.setProperty(DTLoggerFactory.DT_LOGGERS_LEVEL, "com.datatorrent.stram.client.*:INFO,com.datatorrent.stram.api.*:DEBUG");
-    Field f = DTLoggerFactory.class.getDeclaredField("initialized");
-    f.setAccessible(true);
-    f.set(DTLoggerFactory.getInstance(), false);
-    DTLoggerFactory.getInstance().initialize();
+    logger.debug("Logger repository before LoggerUtil.changeLoggersLevel() {}", LogManager.getLoggerRepository());
+    LoggerUtil.changeLoggersLevel(Maps.<String, String>newHashMap());
+    logger.debug("Logger repository after LoggerUtil.changeLoggersLevel() {}", LogManager.getLoggerRepository());
   }
 
   @Test
-  public void test()
+  public void testGetPatternLevels()
   {
-    LoggerFactory.getLogger(DTConfiguration.class);
-    LoggerFactory.getLogger(StramEvent.class);
-    LoggerFactory.getLogger(StreamingAppMaster.class);
-
-    org.apache.log4j.Logger dtConfigLogger = LogManager.getLogger(DTConfiguration.class);
-    Assert.assertEquals(dtConfigLogger.getLevel(), Level.INFO);
-
-    org.apache.log4j.Logger stramEventLogger = LogManager.getLogger(StramEvent.class);
-    Assert.assertEquals(stramEventLogger.getLevel(), Level.DEBUG);
-
-    org.apache.log4j.Logger streamingAppMasterLogger = LogManager.getLogger(StreamingAppMaster.class);
-    Assert.assertNull(streamingAppMasterLogger.getLevel());
+    Map<String, String> changes = Maps.newHashMap();
+    changes.put("_.org.apache.*", "WARN");
```

Comment:
"_." is to avoid impacting log level during unit tests. There is no change in how log levels are set during run-time.